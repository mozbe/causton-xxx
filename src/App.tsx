import { useEffect } from 'react';
import { useRoutes } from 'react-router-dom';
import { getAuth } from 'firebase/auth';
import { initializeFirestore } from 'firebase/firestore';
import { AuthProvider, useInitFirestore, FirestoreProvider, useFirebaseApp } from 'reactfire';
import { ErrorBoundary } from 'react-error-boundary';
import { useRecoilValue, useSetRecoilState } from 'recoil';
import { GlobalStyles, ThemeProvider } from '@mui/material';
import { SnackbarProvider } from 'notistack';
/* *** */
import { routes } from '@routes/routes';
import { loadingState } from '@state/loading';
import { sidebarState } from '@state/sidebar';
import { globals } from '@styles/globals';
import { theme } from '@styles/theme';
import ErrorFallback from '@components/ErrorFallback';
import Sidebar from '@components/Sidebar';

const globalStyles = <GlobalStyles styles={globals} />;

function App() {
  const app = useFirebaseApp();
  const auth = getAuth(app);
  const routeRoutes = useRoutes(routes);
  const sidebar = useRecoilValue(sidebarState);
  const setLoading = useSetRecoilState(loadingState);

  const { status, data: firestoreInstance } = useInitFirestore(async (firebaseApp) => {
    const db = initializeFirestore(firebaseApp, {});
    return db;
  });

  const ready = status === 'success';

  useEffect(() => {
    setLoading(true);

    if (ready) {
      setLoading(false);
    }
  }, [status]);

  if (!ready) return null;

  return (
    <AuthProvider sdk={auth}>
      <SnackbarProvider anchorOrigin={{ vertical: 'top', horizontal: 'right' }}>
        <FirestoreProvider sdk={firestoreInstance}>
          <ThemeProvider theme={theme}>
            {globalStyles}
            <ErrorBoundary FallbackComponent={ErrorFallback}>
              {routeRoutes}
            </ErrorBoundary>
            <Sidebar active={sidebar.active} current={sidebar.current} />
          </ThemeProvider>
        </FirestoreProvider>
      </SnackbarProvider>
    </AuthProvider>
  );
}

export default App;
