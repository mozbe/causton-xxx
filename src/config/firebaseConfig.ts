export const firebaseConfig = {
  apiKey: 'AIzaSyDsYm1LsbTazEwALZiYmBRHrWuCmes_xI8',
  authDomain: 'causton-17012.firebaseapp.com',
  databaseURL: 'https://causton-17012.firebaseio.com',
  projectId: 'causton-17012',
  storageBucket: 'causton-17012.appspot.com',
  messagingSenderId: '45082631698',
  appId: '1:45082631698:web:579480502682fb141902cb',
};
