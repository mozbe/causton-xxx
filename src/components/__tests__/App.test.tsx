import renderer from 'react-test-renderer';
import App from '../../App';
import { FirebaseAppProvider } from 'reactfire';
import { RecoilRoot } from 'recoil';
import { firebaseConfig } from '@config/firebaseConfig';

it('App renders correctly', () => {
  const tree = renderer.create(
    <RecoilRoot>
      <FirebaseAppProvider firebaseConfig={firebaseConfig}>
        <App />
      </FirebaseAppProvider>
    </RecoilRoot>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
