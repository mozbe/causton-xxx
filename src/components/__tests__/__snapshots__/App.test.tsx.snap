// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
<div>
  <mock-styled.h1>
    I am the app
  </mock-styled.h1>
</div>
`;
